redpanda/templates/statefulset.yaml

```diff
@@ -81,21 +84,34 @@ spec:
       containers:
         - name: {{ template "redpanda.name" . }}
           image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          env:
+            - name: SERVICE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
           args:
             - >
-              --check=false
-              --smp {{ .Values.statefulset.resources.limits.cpu }}
-              --memory {{ template "redpanda.parseMemory" . }}
+              redpanda
               start
+              --smp={{ .Values.statefulset.resources.limits.cpu }}
+              --memory={{ template "redpanda.parseMemory" . }}
+              --reserve-memory=0M
+              --advertise-kafka-addr=internal://{{ template "redpanda.kafka.internal.advertise.address" . }}:{{ template "redpanda.kafka.internal.advertise.port" . }}
+              --kafka-addr=internal://{{ template "redpanda.kafka.internal.listen.address" . }}:{{ template "redpanda.kafka.internal.listen.port" . }}
```

Comment:
What is `internal`?

Reply:
It's an arbitrary name that refers to the listener for access internal to K8s, #16 adds `external` for the `LoadBalancer`.

Reply:
It corresponds to the name in rpk's `NamedSocketAddress`?

Reply:
This is the format for passing namedSocketAddress on the command line to `rpk redpanda start`, and I beleive also the format Kafka uses.

Reply:
Thanks, not a Helm expert here, but changes look good. We just need to wait for the updated redpanda image.